[DOCS] DSB-64 removes outdated v2/v3 references from the docs

Rachel-Reverie opened the pull request:
Changes proposed in this pull request:
- Updates "Legacy V2 documentation" and "V3 and newer" in "layout.html" (banner link)
- Removes V2/V3 references from current docs

### Definition of Done
- [X] I have made corresponding changes to the documentation
- [X] I have run any local integration tests and made sure that nothing is broken.

Rachel-Reverie committed:
DOC-324
- Updates "Legacy V2 documentation" and "V3 and newer" in "layout.html" (banner link)

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=7519&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

#### 

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/7519/e82164e1/24eacbb30e7457af3f41ed9faffc061a569a163f.svg)](https://app.codesee.io/r/reviews?pr=7519&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

Rachel-Reverie committed:
DOC-324
- Updates "Legacy V2 documentation" and "V3 and newer" in "layout.html" (banner link)

Rachel-Reverie committed:
DOC-324
- Updates "Legacy V2 documentation" and "V3 and newer" in "layout.html" (banner link)

Rachel-Reverie committed:
DOC-501 - updates not yet migrated warning to reference GX release version rather than API version.

Rachel-Reverie committed:
DOC-500 removes V3 reference in prerequisites.

Rachel-Reverie committed:
DOC-499 Removes V2/V3 references in component files of "How to create and edit Expectations with instant feedback from a sample Batch of data"

Rachel-Reverie committed:
DOC-499 Removes V2/V3 references in component files of "How to create and edit Expectations with instant feedback from a sample Batch of data"

Rachel-Reverie committed:
DOC-498 Removes V3 references from "How to configure a runtime Dataconnector"

Rachel-Reverie committed:
DOC-497 Removes V3 references from "How to choose which Dataconnector to use"

Rachel-Reverie committed:
DOC-496 Removes V3 references from "How to choose which Dataconnector to use"

Rachel-Reverie committed:
DOC-495 Removes V3 references from "How to use GX with GCP and BigQuery"

Rachel-Reverie committed:
DOC-494 Removes V3 references from "How to use GX with airflow"

Rachel-Reverie committed:
fixes minor typo

Rachel-Reverie committed:
DOC-489 Updates V2/V3 references in "Things to Consider" section of "Integration Datahub" guide

Rachel-Reverie committed:
DOC-490 Removes V2/V3 references in the "Data Connector" technical term page

Rachel-Reverie committed:
DOC-495 removes admonition box entirely

Rachel-Reverie committed:
DOC-487 Removes V2 from "How to configure an Expectation Store in Azure Blob Storage"

Rachel-Reverie committed:
Merge branch 'develop' into d/dsb-64/clear_v2_v3_references_in_docs
